pkg/discovery/discovery.go

```diff
@@ -219,3 +229,87 @@ func (d *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(mastersArr, ",")), nil
 }
+
+func (d *tidbDiscovery) VerifyPDEndpoint(advertisePeerURL string) (string, error) {
+	// save a copy of AdvertisePeerURL for failure
+	advertisePeerURL = strings.Replace(advertisePeerURL, "\n", "", -1)
+	copyAdvertisePeerURL := advertisePeerURL
+	pdEndpoint := parseAdvertisePeerURL(advertisePeerURL)
+
+	ns := os.Getenv("MY_POD_NAMESPACE")
+	tc, err := d.cli.PingcapV1alpha1().TidbClusters(ns).Get(pdEndpoint.tcName, metav1.GetOptions{})
+	if err != nil {
+		return advertisePeerURL, err
+	}
+
+	if pdEndpoint.noSchema {
+		if tc.IsTLSClusterEnabled() {
+			pdEndpoint.schema = "https"
+		} else {
+			pdEndpoint.schema = "http"
+		}
+		advertisePeerURL = fmt.Sprintf("%s://%s", pdEndpoint.schema, advertisePeerURL)
+	}
+
+	if pdEndpointHealthCheck(d, tc, advertisePeerURL, pdEndpoint.pdMemberName) {
+		if pdEndpoint.noSchema {
+			return fmt.Sprintf("%s:%s", pdEndpoint.pdMemberName, pdEndpoint.pdMemberPort), nil
+		}
+		return advertisePeerURL, nil
+	}
+
+	for _, pdMember := range tc.Status.PD.PeerMembers {
+		if pdEndpointHealthCheck(d, tc, pdMember.ClientURL, pdMember.Name) {
+			if pdEndpoint.noSchema {
+				return fmt.Sprintf("%s:%s", pdMember.Name, pdEndpoint.pdMemberPort), nil
+			}
+			return pdMember.ClientURL, nil
+		}
+	}
+
+	// if failed, we should return the default value here
+	return copyAdvertisePeerURL, nil
+}
+
+// pdEndpointHealthCheck checks if PD PeerEndpoint is working
+func pdEndpointHealthCheck(d *tidbDiscovery, tc *v1alpha1.TidbCluster, advertisePeerURL string, peerName string) bool {
+	pdClient := d.pdControl.GetPeerPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled(), advertisePeerURL, peerName)
+	_, err := pdClient.GetHealth()
+	return err == nil
+}
+
+// parseAdvertisePeerURL parses advertisePeerURL to PDEndpoint related information
+func parseAdvertisePeerURL(advertisePeerURL string) *pdEndpointURL {
```

Comment:
please add an example url and the corresponding parsed `pdEndpointURL` in comment